### PR TITLE
Add flexible VM sizes support for Azure provider

### PR DESCRIPTION
**Pull request by bpradipt**

### CODE REVIEW

**Comment by bpradipt** [comment]
This is based on the initial PR for AWS #321. The difference is in the parameter names. Instead of instance-type Azure uses instance-sizes. The logic is same as for AWS.

**Review by yoheiueda — APPROVED** [review]
LGTM. Thank you very much!

File: pkg/adaptor/cloud/azure/provider.go [hunk 1]
```diff
@@ -358,3 +367,52 @@ func (p *azureProvider) deleteNetworkInterface(ctx context.Context, nicName stri
...
+
+	// Iterate over the page and populate the instanceSizeSpecList for all the instanceSizes
+	for pager.More() {
+		nextResult, err := pager.NextPage(context.Background())
+		if err != nil {
+			return fmt.Errorf("getting next page of VM sizes: %w", err)
+		}
+		for _, vmSize := range nextResult.VirtualMachineSizeListResult.Value {
+			if util.Contains(instanceSizes, *vmSize.Name) {
+				instanceSizeSpecList = append(instanceSizeSpecList, cloud.InstanceTypeSpec{InstanceType: *vmSize.Name, VCPUs: int64(*vmSize.NumberOfCores), Memory: int64(*vmSize.MemoryInMB)})
```
Comment (snir911) [inline pkg/adaptor/cloud/azure/provider.go:409]: Sorry if it has been discussed before, what if multiple instance types matches same {vCPU, Mem} set?

**Review by snir911 — APPROVED** [review]
Added one question, overall LGTM though

File: pkg/adaptor/cloud/azure/provider.go (hunk 1 above)
Comment (bpradipt) [reply pkg/adaptor/cloud/azure/provider.go:409]: The first one in the list should get selected. We can try this scenario by updating the unit tests of the common cloud function to select bestfit instance type.

**Pull request merged**